src/main/java/org/cloudburstmc/server/registry/BlockRegistry.java

```diff
@@ -61,9 +91,13 @@ private synchronized void registerVanilla(Identifier id, BlockBehavior behavior,
         checkNotNull(id, "id");
         checkNotNull(behavior, "behavior");
         checkNotNull(serializer, "serializer");
-        checkNotNull(traits, "traits");
+        if (traits == null) {
+            traits = new BlockTrait[0];
+        }
         checkClosed();
-        if (this.behaviorMap.containsKey(id)) throw new RegistryException(id + " is already registered");
+
+        if (this.behaviorMap.putIfAbsent(id, behavior) != null) throw new RegistryException(id + " is already registered");
```

Comment:
Why it's checking putIfAbsent ? this will be always not null, right? :/

Reply:
`putIfAbsent` returns the current mapping if the key already exists, so it's an easy way to see if something is registered and place it in the map if not at the same time

Reply:
it'll only be non-null if `id` is already present in the map. this is correct and not an issue